agent/api/statechange.go

```diff
@@ -255,6 +267,33 @@ func (c *ContainerStateChange) String() string {
 	return res
 }
 
+// ToECSAgent converts the agent module level ContainerStateChange to ecs-agent module level ContainerStateChange.
+func (c *ContainerStateChange) ToECSAgent() (*ecs.ContainerStateChange, error) {
+	pl, err := buildContainerStateChangePayload(*c)
+	if err != nil {
+		logger.Error("Could not convert agent container state change to ecs-agent container state change",
+			logger.Fields{
+				"agentContainerStateChange": c.String(),
+				field.Error:                 err,
+			})
+		return nil, err
+	} else if pl == nil {
+		return nil, nil
+	}
+
+	return &ecs.ContainerStateChange{
+		TaskArn:         c.TaskArn,
+		RuntimeID:       aws.StringValue(pl.RuntimeId),
+		ContainerName:   c.ContainerName,
+		Status:          c.Status,
+		ImageDigest:     aws.StringValue(pl.ImageDigest),
+		Reason:          aws.StringValue(pl.Reason),
+		ExitCode:        utils.Int64PtrToIntPtr(pl.ExitCode),
```

Comment:
Why would we be using an int64 for the exit code in the first place?

Reply:
This is due to how FE model defines type `ContainerStateChange`, see: https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/api/ecs/model/ecs/api.go#L9670

`pl` is of the aforementioned type and the aforementioned type's exit code struct field is of type `*int64`.